### PR TITLE
Add ansible lint tool, fix reported issues

### PR DESCRIPTION
**Pull request by sehqlr**
I installed ansible-lint to my mac and ran the linter for all the playbooks in the project. This PR adds the line for ansible-lint in requirements.txt, and begins to make the project compliant with the linter.

A big chunk of the linter errors were about trailing whitespace: I have taken care of those issues in the first commit. I have also made comments on lines where other warnings occurred, but the use of the `command` module here is because the main modules can't do what we need to do in those playbooks. 

The only other error is for `vault_init_json` string being broken up on multiple lines. This is a 'templating' issue.

### CODE REVIEW

File: playbooks/show-package-drift.yml [hunk 1]
```diff
@@ -23,11 +23,11 @@
         state: absent
       with_items: files.stdout_lines
       when: files.stdout_lines|count > 0
-      delegate_to: localhost 
+      delegate_to: localhost
       run_once: yes
 
     - name: get list of packages installed on host
-      shell: rpm -qa | sort -V
+      shell: rpm -qa | sort -V # ansible lint does not like this, wants to use yum module
```
Comment (siddharthist) [inline playbooks/show-package-drift.yml]: Can we change which lint rules are applied, and exclude this one? From ansible-lint's readme:
```
  -x SKIP_LIST          only check rules whose id/tags do not match these
                        values
```
Or perhaps we could go so far as to write our own list of rules (see: @eirslett's comment on #944)?
```
  -r RULESDIR           add one or more rules directories using one or more -r
                        arguments. Defaults to ['/Users/icordasc/sandbox
                        /ansible-lint/lib/ansiblelint/rules'] but any -r flags
                        completely override this.
```

**Comment by sehqlr** [comment]
We can skip the 'resources' tag for linting, and that will take care of those module warnings. I think that adding in the additional rules should be other PRs.

**Comment by sehqlr** [comment]
Also, there's an exception that ansible-lint throws when it loads in a playbook without tasks, such as `terraform.sample.yml`. I created an issue upstream, willthames/ansible-lint#90.

**Comment by willthames** [comment]
I would recommend using 
```
tags: 
- skip_ansible_lint
```
for this scenario, where you know you're using `rpm` properly but don't want to turn off the checks across the board.

**Comment by siddharthist** [comment]
Thanks @willthames! That sounds like a great idea.

File: requirements.txt [hunk 2]
```diff
@@ -3,3 +3,4 @@ ansible>=1.9.1,<2.0
 pytest==2.7.1
 PyYAML==3.11
 Sphinx==1.2.3
+ansible-lint
```
Comment (siddharthist) [inline requirements.txt]: We should always fix some kind of version requirement, just in case another project makes backwards-incompatible changes.

**Comment by siddharthist** [comment]
Works for me!
```
$ find roles -name "*.yml" -print0 | xargs -0 ansible-lint
$ ansible-lint kubernetes.yml 
$ ansible-lint terraform.sample.yml 
$ find playbooks -name "*.yml" -print0 | xargs -0 ansible-lint
```

**Pull request merged**